cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerHostStatusChecker.java

```diff
@@ -48,6 +41,31 @@ protected boolean doStatusCheck(ClouderaManagerPollerObject pollerObject, Comman
         }
     }
 
+    private List<InstanceMetaData> collectNotKnownInstancesByManager(ClouderaManagerPollerObject pollerObject, List<String> hostIpsFromManager) {
+        return pollerObject.getStack().getInstanceMetaDataAsList().stream()
+                    .filter(metaData -> metaData.getDiscoveryFQDN() != null)
+                    .filter(metaData -> !metaData.isTerminated() && !metaData.isDeletedOnProvider())
+                    .filter(metaData -> !hostIpsFromManager.contains(metaData.getPrivateIp()))
+                    .collect(Collectors.toList());
+    }
+
+    private List<String> fetchHeartbeatedHostIpsFromManager(ClouderaManagerPollerObject pollerObject) throws ApiException {
+        HostsResourceApi hostsResourceApi = new HostsResourceApi(pollerObject.getApiClient());
+        String viewType = "FULL";
+        ApiHostList hostList = hostsResourceApi.readHosts(viewType);
+        List<String> hostIpsFromManager = filterForHeartBeatedIps(hostList);
+        LOGGER.debug("Hosts in the list from manager: " + hostIpsFromManager);
+        return hostIpsFromManager;
+    }
+
+    private List<String> filterForHeartBeatedIps(ApiHostList hostList) {
+        return hostList.getItems().stream()
+                    .filter(item -> StringUtils.isNotBlank(item.getLastHeartbeat()))
+                    .filter(item -> now.isBefore(Instant.parse(item.getLastHeartbeat())))
```

Comment:
Instead of now I would call it start as it's kind of misleading.

Reply:
sure, renamed